### PR TITLE
Feat/instance metadata options

### LINKED ISSUE #87: Expose metadata_options {instance_metadata_tags}
## Describe the Feature

For some time AWS allows to enable instance tags in metadata through:

```hcl
  metadata_options {
    instance_metadata_tags = "enabled"
  }
```

See: https://aws.amazon.com/about-aws/whats-new/2022/01/instance-tags-amazon-ec2-instance-metadata-service/

## Expected Behavior

Ability to enable instance_metadata_tags

## Use Case

This is very hand for setting up a hostname from "Tag:Name" and other useful automations without AWS SDK installed e.g. following cloud init example old way:

```sh
set -x
r=$(exec curl -s http://169.254.169.254/latest/dynamic/instance-identity/document/ | jq -r .region)
i=$(exec curl -s http://169.254.169.254/latest/dynamic/instance-identity/document/ | jq -r .instanceId)
n=$(aws ec2 --region "$r" describe-instances --instance-ids "$i" | jq -r '.Reservations[].Instances[].Tags[] | select(.Key=="Name") | .Value')
if [[ -n "$n" ]]; then
    hostnamectl set-hostname --static "$n" 
fi
```

Simpler way:

```sh
set -x

n=$(curl http://169.254.169.254/latest/meta-data/tags/instance)
if [[ -n "$n" ]]; then
    hostnamectl set-hostname --static "$n" 
fi
```

## Describe Ideal Solution

Just add it to the TF ;)

## Alternatives Considered

There are no alternatives.

### PR DESCRIPTION
**Pull request by zen**
## what
* Added support for newly introduced instance tags in metadata

## why
* This is very handy for setting up a hostname from "Tag:Name" and other useful automations without AWS SDK installed

## references
* https://aws.amazon.com/about-aws/whats-new/2022/01/instance-tags-amazon-ec2-instance-metadata-service/
* closes #87

### CODE REVIEW

**Comment by zen** [comment]
@adamcrews I would appreciate review here, I'd like to go further and add the same support to https://github.com/cloudposse/terraform-aws-eks-workers

File: variables.tf [hunk 1]
```diff
@@ -465,6 +465,12 @@ variable "metadata_http_protocol_ipv6_enabled" {
   description = "Set true to enable IPv6 in the launch template."
 }
 
+
+variable "metadata_instance_metadata_tags_enabled" {
+  type        = bool
+  default     = false
+  description = "Set true to enable metadata tags in the launch template."
+}
```
Comment (Nuru) [inline variables.tf:473]:
```suggestion
variable "metadata_instance_metadata_tags_enabled" {
  type        = bool
  default     = false
  description = "Set true to enable metadata tags in the launch template."
}

```

**Review by Nuru — CHANGES_REQUESTED** [review]
@zen Thank you for the contribution, and apologies for taking so long to get to it. Please make the trivial whitespace change I noted, and copy `versions.tf` from the top level to `examples/complete/`. That should fix the test failures. If the tests succeed, I will approve the PR.

**Comment by jamengual** [comment]
closed by https://github.com/cloudposse/terraform-aws-ec2-autoscale-group/pull/94

**Pull request closed without merging**